### PR TITLE
Fix functional tests

### PR DESCRIPTION
**Pull request by pakrym**
After https://github.com/aspnet/Hosting/pull/435 
@rynowak @davidfowl

### CODE REVIEW

**Comment by rynowak** [comment]
If you do a search, I think there are two other places where CallContextSP is used in MVC

File: test/Microsoft.AspNet.Mvc.FunctionalTests/TestHelper.cs [hunk 1]
```diff
@@ -60,7 +60,7 @@ public static TestServer CreateServer(Action<IApplicationBuilder> builder, strin
 
             // Get current IApplicationEnvironment; likely added by the host.
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/TestHelper.cs]: This comment isn't relevant anymore

File: test/Microsoft.AspNet.Mvc.FunctionalTests/TestHelper.cs [hunk 2]
```diff
@@ -58,9 +59,8 @@ public static TestServer CreateServer(Action<IApplicationBuilder> builder, strin
         {
             applicationPath = applicationPath ?? WebsitesDirectoryPath;
 
-            // Get current IApplicationEnvironment; likely added by the host.
             var provider = services.BuildServiceProvider();
-            var originalEnvironment = provider.GetRequiredService<IApplicationEnvironment>();
+            var originalEnvironment = PlatformServices.Default.Application;
```
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/TestHelper.cs]: We get `originalEnvironment` to wrap it in our `TestApplicationEnvironment`. Will the `TestApplicationEnvironment` instance added just below be picked up correctly?

File: test/Microsoft.AspNet.Mvc.FunctionalTests/TestHelper.cs [hunk 3]
```diff
@@ -47,7 +47,8 @@ public static TestServer CreateServer(Action<IApplicationBuilder> builder, strin
         {
             return TestServer.Create(
                 builder,
-                services => AddTestServices(services, applicationWebSiteName, applicationPath, configureServices));
+                configureServices: null,
+                configureHostServices: services => AddTestServices(services, applicationWebSiteName, applicationPath, configureServices));
```
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/TestHelper.cs]: Harder to read here. Please wrap the parameters to their own lines.

**Comment by dougbu** [comment]
+1 to @rynowak's comment. Please fix [`DefaultAssemblyProviderTests`](https://github.com/aspnet/Mvc/blob/dev/test/Microsoft.AspNet.Mvc.Core.Test/Infrastructure/DefaultAssemblyProviderTests.cs#L256) as well

File: test/Microsoft.AspNet.Mvc.FunctionalTests/TestHelper.cs (hunk 2 above)
Comment (pakrym) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/TestHelper.cs]: Yes, new environment will be picked up, it is executed in WebHostBuilder.BuildHostingServices and default services are added using TryAdd so it won't be overwritten

File: test/Microsoft.AspNet.Mvc.FunctionalTests/TestHelper.cs (hunk 3 above)
Comment (pakrym) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/TestHelper.cs]: What do you mean?

AddTestServices(
services, 
applicationWebSiteName, applicationPath, configureServices)

Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/TestHelper.cs]: Yes. With relative indentation, it should look like:
``` c#
return TestServer.Create(
    builder,
    configureServices: null,
    configureHostServices: services => AddTestServices(
        services,
        applicationWebSiteName,
        applicationPath,
        configureServices));
```

File: test/Microsoft.AspNet.Mvc.Core.Test/Infrastructure/DefaultAssemblyProviderTests.cs [hunk 4]
```diff
@@ -253,8 +253,7 @@ private static IEnumerable<string> GetAssemblies(IEnumerable<string> libraries)
 
             private static ILibraryManager GetLibraryManager()
             {
-                return CallContextServiceLocator.Locator.ServiceProvider
-                    .GetRequiredService<ILibraryManager>();
+                return PlatformServices.Default.LibraryManager;
```
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.Core.Test/Infrastructure/DefaultAssemblyProviderTests.cs]: thx

**Comment by dougbu** [comment]
:shipit: once you're able to verify the complete build succeeds.

**Comment by davidfowl** [comment]
:shipit:

**Pull request merged**